Add README and CONTRIBUTING docs. (#261)
melink14/rikaikun: CONTRIBUTING.md

# Contributing <!-- omit in toc -->

- [Introduction](#introduction)
- [Ground Rules](#ground-rules)
- [Code Contributions](#code-contributions)
  - [Your First Code Contribution](#your-first-code-contribution)
  - [Getting started](#getting-started)
  - [Pull Requests](#pull-requests)
- [Style Guides](#style-guides)
  - [Git Commit Messages](#git-commit-messages)
  - [Coding style](#coding-style)
    - [Style preferences not handled by the linter](#style-preferences-not-handled-by-the-linter)

# Introduction

> First, thank you for considering contributing! It's people like you that make
> the open source community such a great community! 😊

Reviewer: ```suggestion
Thank you for considering contributing! It's people like you that make
the open source community so great! 😊
```

- Usually, your PR should start with one commit which contains your new code. During the course of review, it's better to add commits to your branch instead of amending or rebasing because otherwise Github loses the evolution of the code. When your PR is merged, it will be squashed back into one commit.

# Style Guides

## Git Commit Messages

rikaikun follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification. This allows us to automatically generate new versions and change logs and reduces some cognitive load when first reading a commit.

In brief, commit messages should look like:

```
<type>[optional scope]: <description>

[optional body]

[optional footer(s)]
```

If your change requires a newer version of chrome than was required previously, include `BREAKING CHANGE` with an explanation in the footer.

Here are examples of the types currently in use ([source](http://karma-runner.github.io/1.0/dev/git-commit-msg.html)):

```
chore: add Oyster build script
docs: explain hat wobble
feat: add beta sequence
fix: remove broken confirmation message
refactor: share logic between 4d3d3d3 and flarhgunnstow
style: convert tabs to spaces

> test: ensure Tayne retains clothing

Reviewer: 😳

```

These are the common scopes used, though feel free to suggest a new one if it makes sense:
| Scope | Explanation |
|-------|------------------------------------------------------------|
| deps | Used when changing the node dependencies of the project. |
| ui | Used for fixes and features affecting the popup UI. |
| dict | Used for fixes and features affecting the dictionary data. |

See the (recent) [commit history](https://github.com/melink14/rikaikun/commits/master) for more examples of correctly formatted messages.

## Coding style

> For now as long as `npm run lint` passes, your code is correctly styled. You may get some more specific advice if a maintainer thinks something is more readable but in general there's no use arguing about that which a robot can handle.

Author: Made it more positive.

# Contributing <!-- omit in toc -->

- [Introduction](#introduction)
- [Ground Rules](#ground-rules)
- [Code Contributions](#code-contributions)
  - [Your First Code Contribution](#your-first-code-contribution)
  - [Getting started](#getting-started)
  - [Pull Requests](#pull-requests)
- [Style Guides](#style-guides)
  - [Git Commit Messages](#git-commit-messages)
  - [Coding style](#coding-style)
    - [Style preferences not handled by the linter](#style-preferences-not-handled-by-the-linter)

# Introduction

Thank you for considering contributing! It's people like you that make the open source community so great! 😊

The tips and guidelines below are intended to help both contributors and maintainers respect each other's time and effect change as quickly as possible.

> - **Bugs and Feature Requests**: Follow the templates [here](https://github.com/melink14/rikaikun/issues/new/choose) to report problems or ideas. Don't be surprised if you're asked for clarification or opinions on solutions.

Reviewer: > Don't be surprised if you're asked for clarification or opinions on solutions.

This comes off a little strong. Maybe something like "you may be asked to share..."

Author: will consider; test.

Author: test2

> - **Bugs and Feature Requests**: Follow the templates [here](https://github.com/melink14/rikaikun/issues/new/choose) to report problems or ideas. Don't be surprised if you're asked for clarification or opinions on solutions.

Author: I went with 'you may be asked for...'

> - **Code**: take a look at the open issues. Even if you can't write code, commenting on them, showing that you care about a given issue matters. It helps us triage them.

Reviewer: This section talks more about issues than code. Should it be renamed?

```suggestion
- **Code**: Take a look at the open issues. Even if you are unable to implement a solution, commenting on issues with your opinions and experiences is greatly appreciated and helpful for triage.```

Author: maybe Implementation? Maybe Fixes?

> - **Documentation**: Update or add a page to the [wiki](https://github.com/melink14/rikaikun/wiki) or improve some already existing documentation to help other users.

Reviewer: ```suggestion
- **Documentation**: Update or add a page to the [wiki](https://github.com/melink14/rikaikun/wiki) or improve preexisting documentation to help other users.
```

Author: Sounds good.

> - **Something Else?**: This list isn't conclusive so if you have another idea for contribution, please reach out.

Reviewer: ```suggestion
- **Something Else?**: This list isn't comprehensive, so if you have another idea for contribution, please reach out.
```

Author: :thumbsup:

# Ground Rules

**Responsibilities**

- Be welcoming and treat everyone with respect as per the [Code of Conduct](CODE_OF_CONDUCT.md).
- Before doing any major work, [open an issue](https://github.com/melink14/rikaikun/issues/new/choose) to garner community feedback.

# Code Contributions

## Your First Code Contribution

Working on your first pull request? You can learn how from this _free_ series,
[How to Contribute to an Open Source Project on GitHub](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).

[Here are some good first issues.](https://github.com/melink14/rikaikun/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22)

> If you get stuck push your code early and ask for feedback on the issue.

Reviewer: ```suggestion
If you get stuck, push your code early and ask for feedback on the issue.
```

Author: test

Author: :thumbsup:

## Getting started

> 1. After you check out the code, run `npm install` to get the rikaikun dev tools ready.

Author: Add instructions on installing npm?

2. Make your changes and commit them locally. Run `npm run fix` often to ensure your code follows style guidelines.
3. Run `npm run build` to create an unpackaged instance of rikaikun in the `dist` directory. Load this into Chrome to test your changes.

> 4. When you're satisfied, commit you're code (make sure to follow the commit message guidelines) and start a pull request.

Reviewer: ```suggestion
4. When you're satisfied with your changes, commit your code, check that your commit message follows [these guidelines](#git-commit-messages), and start a pull request.
```

Author: :thumbsup:
